components/Banner/src/MDCBannerView.m

```diff
@@ -0,0 +1,474 @@
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *leadingButtonConstraintCenterY;
+@property(nonatomic, readwrite, strong)
+    NSLayoutConstraint *leadingButtonConstraintTrailingWithTrailingButton;
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *trailingButtonConstraintBottom;
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *trailingButtonConstraintTop;
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *trailingButtonConstraintTrailing;
+
+@end
+
+@implementation MDCBannerView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (void)commonBannerViewInit {
+  self.backgroundColor = UIColor.whiteColor;
+
+  // Create textLabel
+  UILabel *textLabel = [[UILabel alloc] init];
+  textLabel.translatesAutoresizingMaskIntoConstraints = NO;
+  textLabel.font = [MDCTypography body2Font];
+  textLabel.textColor = UIColor.blackColor;
+  textLabel.alpha = [MDCTypography body2FontOpacity];
+  textLabel.numberOfLines = kTextNumberOfLineLimit;
+  textLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+  [self addSubview:textLabel];
+  _textLabel = textLabel;
+
+  // Create imageView
+  UIImageView *imageView = [[UIImageView alloc] init];
+  imageView.translatesAutoresizingMaskIntoConstraints = NO;
+  if (@available(iOS 9.0, *)) {
+    [imageView.widthAnchor constraintEqualToConstant:kImageViewSideLength].active = YES;
+    [imageView.heightAnchor constraintEqualToConstant:kImageViewSideLength].active = YES;
+  }
+  imageView.contentMode = UIViewContentModeCenter;
+  imageView.clipsToBounds = YES;
+  imageView.hidden = YES;
+  [self addSubview:imageView];
+  _imageView = imageView;
+
+  // Create a button container to organize buttons
+  UIView *buttonContainerView = [[UIView alloc] init];
+  buttonContainerView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self addSubview:buttonContainerView];
+  self.buttonContainerView = buttonContainerView;
+
+  // Create leadingButton and trailingButton
+  MDCButton *leadingButton = [[MDCButton alloc] init];
+  leadingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  leadingButton.backgroundColor = UIColor.whiteColor;
+  [buttonContainerView addSubview:leadingButton];
+  _leadingButton = leadingButton;
+
+  MDCButton *trailingButton = [[MDCButton alloc] init];
+  trailingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  trailingButton.backgroundColor = UIColor.whiteColor;
+  [buttonContainerView addSubview:trailingButton];
+  _trailingButton = trailingButton;
+
+  [self addObserversToSubviews];
+  [self setupConstraints];
+}
+
+#pragma mark - Constraints Helpers
+
+- (void)setupConstraints {
```

Comment:
Mark this method with `NS_AVAILABLE_IOS(9_0)` so that we can remove a layer of indentation.

Reply:
Addressed in aeeecd3982520bfa91577f0db37cb0e67005d986.

Reply:
Ah this one wasn't resolved. To arm you with the underlying motivation here:

Any private method that solely consists of logic nested within an `if (@available(` statement and is not a subclass override should be annotated with NS_AVAILABLE_IOS instead. This allows us to remove a layer of logical indication and more clearly communicate that the method itself is only available for a given platform.

With the above definition you should also not need to both annotate a method and include the runtime check, because by definition only this class is expected to invoke these methods and the corresponding call-sites can be correctly annotated with the availability checks.

Reply:
I've filed https://github.com/material-components/material-components-ios/issues/7070 to track this type of feedback so we don't churn on it further.

Reply:
This one was addressed and then changed because of another code review comment (https://github.com/material-components/material-components-ios/pull/7065#discussion_r271731598) later. I was kinda of confused on the convention/guideline here, and I felt either way could work.

Thanks for filing the issue.